drivers/temperature/adt7420/adt7420.c

```diff
@@ -92,38 +90,68 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
- * @param num_data_bytes   - Number of bytes.
- * @param data             - Data to be written in input register.
  * @param mask		   - Bit Mask of the bit to be written
  * @param value		   - Value of the bit
  *
  * @return 0 in case of success, -1 otherwise.
 *******************************************************************************/
-int32_t set_register_value(struct adt7420_dev *dev,
-			   uint8_t register_address,
-			   uint8_t num_data_bytes,
-			   uint8_t *data, uint8_t mask, uint8_t value)
+int adt7420_set_register_mask(struct adt7420_dev *dev,
+			      uint8_t register_address, uint8_t mask, uint8_t value)
 {
 
 	uint8_t data_buffer[3] = {0, 0, 0};
+	uint16_t regval;
+
+	regval = adt7420_get_register_value(dev, register_address);
+
+	regval &= ~mask;
+	regval |= value;
+	if (!adt7420_is_spi(dev) || register_address == ADT7320_REG_STATUS
+	    || register_address == ADT7320_REG_CONFIG || register_address == ADT7320_REG_ID
+	    || register_address == ADT7320_REG_HIST)
+		data_buffer[0] = regval;
+	else {
+		data_buffer[0] = regval >> 8;
+		data_buffer[1] = regval & 0xFF;
+	}
+
+	if (adt7420_reg_write(dev, register_address, data_buffer))
+		return -1;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Sets the value of a register SPI/I2C.
+ *
+ * @param dev              - The device structure.
+ * @param register_address - Address of the register.
+ * @param data             - Data to be written in input register.
+ *
+ * @return 0 in case of success, -1 otherwise.
+*******************************************************************************/
+int adt7420_reg_write(struct adt7420_dev *dev,
+		      uint8_t register_address,
+		      uint8_t *data)
```

Comment:
it's a bit uncommon to pass the data parameter as pointer in the write function. use a normal variable and based on the register size process the data bits.